app/controllers/users_controller.rb

```diff
@@ -1,3 +1,29 @@
 class UsersController < ApplicationController
-  def new; end
+  attr_reader :user
+
+  def show
+    @user = User.find_by(id: params[:id])
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new user_params
+
+    if user.save
+      flash[:success] = t "user.welcome"
+      redirect_to user
+    else
+      render :new
```

Comment:
Vì ở đây em cần load lại html mà vẫn giữ thông tin user để có thể in ra lỗi. Nếu dùng redirect thì nó sẽ load lại cả trang và thông tin user cũng mất luôn nên sẽ không in được lỗi ra